orangecontrib/text/widgets/owimporttexts.py

```diff
@@ -0,0 +1,642 @@
+"""
+Import Texts
+-------------
+
+Import texts 'into' canvas session from a local file system
+"""
+import sys
+import os
+import enum
+import warnings
+import logging
+import traceback
+
+from types import SimpleNamespace as namespace
+
+import numpy as np
+
+from AnyQt.QtCore import Qt, QEvent, QFileInfo, QThread
+from AnyQt.QtCore import pyqtSlot as Slot
+from AnyQt.QtGui import QStandardItem, QDropEvent
+
+from AnyQt.QtWidgets import (
+    QAction, QPushButton, QComboBox, QApplication, QStyle, QFileDialog,
+    QFileIconProvider, QStackedWidget, QProgressBar, QWidget, QHBoxLayout,
+    QVBoxLayout, QLabel
+)
+
+import Orange.data
+
+from Orange.widgets import widget, gui, settings
+from Orange.widgets.utils.filedialogs import RecentPath
+from Orange.widgets.utils.concurrent import (
+    ThreadExecutor, FutureWatcher, methodinvoke
+)
+from Orange.widgets.widget import Output
+
+from orangecontrib.text.corpus import Corpus
+from orangecontrib.text.widgets.utils.readers import TextScan
+from Orange.canvas.preview.previewbrowser import TextLabel
+
+
+def prettifypath(path):
+    home = os.path.expanduser("~/")
+    if path.startswith(home):  # case sensitivity!
+        path = os.path.join("~", os.path.relpath(path, home))
+    return path
+
+
+log = logging.getLogger(__name__)
+
+
+class RuntimeEvent(QEvent):
+    Init = QEvent.registerEventType()
+
+
+def RecentPath_asqstandarditem(pathitem):
+    icon_provider = QFileIconProvider()
+    # basename of a normalized name (strip right path component separators)
+    basename = os.path.basename(os.path.normpath(pathitem.abspath))
+    item = QStandardItem(
+        icon_provider.icon(QFileInfo(pathitem.abspath)),
+        basename
+    )
+    item.setToolTip(pathitem.abspath)
+    item.setData(pathitem, Qt.UserRole)
+    return item
+
+
+class State(enum.IntEnum):
+    NoState, Processing, Done, Cancelled, Error = range(5)
+
+
+class OWImportTexts(widget.OWWidget):
+    name = "Import Texts"
+    description = "Import text document(s) from a directory(s)"
```

Comment:
I also vote for: Import text documents from folders.
Plural is enough and perhaps folder is more intuitive to directory?